lib/api/core/hotelClerk.js

```diff
@@ -391,35 +391,32 @@ class HotelClerk {
           throw new SizeLimitError('Unable to subscribe: maximum number of unique rooms reached');
         }
 
-        const response = this.kuzzle.dsl.store(normalized);
-
-        return this.kuzzle.pluginsManager.trigger('room:new', {index, collection, roomId: response.id})
-          .then(modified => {
-            // pipes are not allowed to modify the room id
-            const roomId = response.id;
-
-            /*
-             In some very rare cases, the room may have been created between
-             the beginning of the function executed at the end of dsl.normalize,
-             and this one
-
-             Before incrementing the rooms count, we have to make sure this
-             is not the case to ensure our counter is right
-             */
-            if (!this.rooms[roomId]) {
-              this.roomsCount++;
-            }
-
-            this.rooms[roomId] = {
-              id: roomId,
-              customers: new Set(),
-              channels: {},
-              index: modified.index,
-              collection: modified.collection
-            };
+        const 
+          response = this.kuzzle.dsl.store(normalized),
+          roomId = response.id;
+
+        this.kuzzle.pluginsManager.trigger('room:new', {index, collection, roomId: response.id});
+
+        /*
+         In some very rare cases, the room may have been created between
+         the beginning of _createRoom and after dsl.normalize finished its job
+
+         Before incrementing the rooms count, we have to make sure this
+         is not the case to ensure our counter is right
+         */
+        if (!this.rooms[roomId]) {
+          this.roomsCount++;
+
+          this.rooms[roomId] = {
+            index,
+            collection,
+            id: roomId,
+            customers: new Set(),
+            channels: {},
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this trailing comma. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ATrailingComma)